### ISSUE TITLE
How to send a JSON array or object as a parameter in a POST request?

### ISSUE THREAD

**Issue by jiguantong** [api3dao/airnode#1790]
## Description

I want to call this API through airnode: [eth_getBlockByNumber](https://www.quicknode.com/docs/arbitrum/eth_getBlockByNumber), it needs a Json array parameter like `"params":["finalized",false]`, I want to know how to set it in the config.json `ois.endpoints.fixedOperationParameters`

I tried to set it in `fixedOperationParameters`
```json
{
                            "operationParameter": {
                                "in": "query",
                                "name": "params"
                            },
                            "value": ["finalized",false]
 }
```
The error message is:
```
Error: Invalid Airnode configuration file: [
  {
    "code": "invalid_type",
    "expected": "string",
    "received": "array",
    "path": [
      "ois",
      0,
      "endpoints",
      0,
      "fixedOperationParameters",
      4,
      "value"
    ],
    "message": "Expected string, received array"
  }
]
```
I also tried to leave it in `parameters` without the value, and fill it in postman when sending the request, the error message is "Invalid request body"
```
[2023-05-31 10:33:38.294] INFO API gateway server running on "http://localhost:3000"                            
[2023-05-31 10:34:01.232] DEBUG HTTP gateway request received                                                    requestId:724f5b1b-9cb5-4af5-8930-88c98ab1e1df
[2023-05-31 10:34:01.232] DEBUG HTTP gateway request passed origin verification                                  requestId:724f5b1b-9cb5-4af5-8930-88c98ab1e1df
[2023-05-31 10:34:01.232] DEBUG HTTP gateway request passed request body parsing                                 requestId:724f5b1b-9cb5-4af5-8930-88c98ab1e1df
[2023-05-31 10:34:01.233] ERROR HTTP gateway request verification error                                          requestId:724f5b1b-9cb5-4af5-8930-88c98ab1e1df
[2023-05-31 10:34:01.296] INFO Created initial coordinator state                                                Coordinator-ID:af35db446c6ced1f
```

## Version

node: v18.16.0
yarn: 1.22.19
airnode: api3/airnode-client:latest

## Present Behaviour

no

## Expected Behaviour

Should accept the parameter and send it to the server.

## Steps to reproduce

1. Copy my config.json
```json
{
    "chains": [
        {
            "authorizers": {
                "requesterEndpointAuthorizers": [],
                "crossChainRequesterAuthorizers": [],
                "requesterAuthorizersWithErc721": [],
                "crossChainRequesterAuthorizersWithErc721": []
            },
            "authorizations": {
                "requesterEndpointAuthorizations": {}
            },
            "contracts": {
                "AirnodeRrp": "0x2ab9f26E18B64848cd349582ca3B55c2d06f507d"
            },
            "id": "11155111",
            "providers": {
                "myChainProvider": {
                    "url": "${LOCAL_PROVIDER_URL}"
                }
            },
            "type": "evm",
            "options": {
                "fulfillmentGasLimit": 500000,
                "gasPriceOracle": [
                    {
                        "gasPriceStrategy": "latestBlockPercentileGasPrice",
                        "percentile": 60,
                        "minTransactionCount": 20,
                        "pastToCompareInBlocks": 20,
                        "maxDeviationMultiplier": 2
                    },
                    {
                        "gasPriceStrategy": "providerRecommendedGasPrice",
                        "recommendedGasPriceMultiplier": 1.2
                    },
                    {
                        "gasPriceStrategy": "providerRecommendedEip1559GasPrice",
                        "baseFeeMultiplier": 2,
                        "priorityFee": {
                            "value": 3.12,
                            "unit": "gwei"
                        }
                    },
                    {
                        "gasPriceStrategy": "constantGasPrice",
                        "gasPrice": {
                            "value": 10,
                            "unit": "gwei"
                        }
                    }
                ]
            },
            "maxConcurrency": 100
        }
    ],
    "nodeSettings": {
        "cloudProvider": {
            "type": "local"
        },
        "airnodeWalletMnemonic": "${AIRNODE_WALLET_MNEMONIC}",
        "heartbeat": {
            "enabled": false
        },
        "httpGateway": {
            "enabled": true,
            "maxConcurrency": 20,
            "corsOrigins": []
        },
        "httpSignedDataGateway": {
            "enabled": false
        },
        "oevGateway": {
            "enabled": false
        },
        "logFormat": "plain",
        "logLevel": "DEBUG",
        "nodeVersion": "0.11.1",
        "stage": "quick-container"
    },
    "triggers": {
        "rrp": [
            {
                "endpointId": "0x6db9e3e3d073ad12b66d28dd85bcf49f58577270b1cc2d48a43c7025f5c27af6",
                "oisTitle": "CoinGecko Basic Request",
                "endpointName": "coinMarketData",
                "cacheResponses": false
            }
        ],
        "http": [
            {
                "endpointId": "0x6db9e3e3d073ad12b66d28dd85bcf49f58577270b1cc2d48a43c7025f5c27af6",
                "oisTitle": "CoinGecko Basic Request",
                "endpointName": "coinMarketData"
            }
        ],
        "httpSignedData": []
    },
    "templates": [],
    "ois": [
        {
            "oisFormat": "2.0.0",
            "title": "CoinGecko Basic Request",
            "version": "1.0.0",
            "apiSpecifications": {
                "servers": [
                    {
                        "url": "https://arbitrum-mainnet.infura.io/v3"
                    }
                ],
                "paths": {
                    "/${MyInfuraKey}": {
                        "post": {
                            "parameters": [
                                {
                                    "in": "header",
                                    "name": "Content-Type"
                                },
                                {
                                    "in": "query",
                                    "name": "method"
                                },
                                {
                                    "in": "query",
                                    "name": "id"
                                },
                                {
                                    "in": "query",
                                    "name": "jsonrpc"
                                },
                                {
                                    "in": "query",
                                    "name": "params"
                                }
                            ]
                        }
                    }
                },
                "components": {
                    "securitySchemes": {}
                },
                "security": {}
            },
            "endpoints": [
                {
                    "name": "coinMarketData",
                    "operation": {
                        "method": "post",
                        "path": "/${MyInfuraKey}"
                    },
                    "fixedOperationParameters": [
                        {
                            "operationParameter": {
                                "in": "header",
                                "name": "Content-Type"
                            },
                            "value": "application/json"
                        },
                        {
                            "operationParameter": {
                                "in": "query",
                                "name": "method"
                            },
                            "value": "eth_getBlockByNumber"
                        },
                        {
                            "operationParameter": {
                                "in": "query",
                                "name": "id"
                            },
                            "value": "1"
                        },
                        {
                            "operationParameter": {
                                "in": "query",
                                "name": "jsonrpc"
                            },
                            "value": "2.0"
                        },
                        {
                            "operationParameter": {
                                "in": "query",
                                "name": "params"
                            },
                            "value": ["finalized",false]
                        }
                    ],
                    "reservedParameters": [
                        {
                            "name": "_type",
                            "fixed": "string"
                        },
                        {
                            "name": "_path",
                            "fixed": "result"
                        }
                    ],
                    "parameters": [
                    ]
                }
            ]
        }
    ],
    "apiCredentials": []
}
```
2. Copy my secrets.env
```
AIRNODE_WALLET_MNEMONIC=""
LOCAL_PROVIDER_URL=""
```
3. Run this command
```
docker run \
  --volume "$(pwd):/app/config" \
  --name quick-start-container-airnode \
  --network host \
  api3/airnode-client:latest
```

**Comment by bbenligiray** [comment]
Hey @jiguantong As the error says, `value` expects a string, so you'll need to enclose its value with quotation marks. Also note that JSON requires you to escape your `"` characters. I believe this should work:

```
{
                            "operationParameter": {
                                "in": "query",
                                "name": "params"
                            },
                            "value": "[\"finalized\",false]"
 }
```

**Comment by jiguantong** [comment]
> Hey @jiguantong As the error says, `value` expects a string, so you'll need to enclose its value with quotation marks. Also note that JSON requires you to escape your `"` characters. I believe this should work:
> 
> ```
> {
>                             "operationParameter": {
>                                 "in": "query",
>                                 "name": "params"
>                             },
>                             "value": "[\"finalized\",false]"
>  }
> ```

Thanks for your help! I have tried it and it returned error code 400. This is the log.
```
[2023-06-01 07:49:25.360] ERROR Failed to call Endpoint:coinMarketData                                           requestId:10739fc2-d150-463f-8846-5b98e30b5bae, Endpoint-ID:0x6db9e3e3d073ad12b66d28dd85bcf49f58577270b1cc2d48a43c7025f5c27af6
[2023-06-01 07:49:25.362] ERROR AxiosError: Request failed with status code 400
    at settle (/usr/local/share/.config/yarn/global/node_modules/axios/dist/node/axios.cjs:1909:12)
    at IncomingMessage.handleStreamEnd (/usr/local/share/.config/yarn/global/node_modules/axios/dist/node/axios.cjs:2989:11)
    at IncomingMessage.emit (node:events:525:35)
    at IncomingMessage.emit (node:domain:489:12)
    at endReadableNT (node:internal/streams/readable:1359:12)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) requestId:10739fc2-d150-463f-8846-5b98e30b5bae, Endpoint-ID:0x6db9e3e3d073ad12b66d28dd85bcf49f58577270b1cc2d48a43c7025f5c27af6
[2023-06-01 07:49:25.363] ERROR HTTP gateway request processing error 
```

I also tried send it to api directly in postman, it returned invalid json request. So I think the API needs Json array, rather than string.
![image](https://github.com/api3dao/airnode/assets/27564714/7a136746-7c9b-42f5-ab41-a38a67f81df2)

**Comment by bbenligiray** [comment]
Alright, let us investigate and get back to you

**Comment by jiguantong** [comment]
> @jiguantong - PRs by @aquarat to address this issue have been merged. We'll let you know when a release is available that incorporates the changes.
Cool! Thanks!

**Comment by dcroote** [comment]
@jiguantong - @aquarat had a good idea in that you can test the changes now using the API3 docker images that are built during CI.

`0b7d89eb582d63aa299216f3cc28d82cf7071981` is the latest master commit. To get the deployer docker image (e.g. for AWS deployment) corresponding to this commit you can use:
```sh
docker pull api3/airnode-deployer-dev:0b7d89eb582d63aa299216f3cc28d82cf7071981
```

Similarly, for the local client image:
```sh
docker pull api3/airnode-client-dev:0b7d89eb582d63aa299216f3cc28d82cf7071981
```

**Comment by jiguantong** [comment]
> @jiguantong - @aquarat had a good idea in that you can test the changes now using the API3 docker images that are built during CI.
> 
> `0b7d89eb582d63aa299216f3cc28d82cf7071981` is the latest master commit. To get the deployer docker image (e.g. for AWS deployment) corresponding to this commit you can use:
> 
> ```shell
> docker pull api3/airnode-deployer-dev:0b7d89eb582d63aa299216f3cc28d82cf7071981
> ```
> 
> Similarly, for the local client image:
> 
> ```shell
> docker pull api3/airnode-client-dev:0b7d89eb582d63aa299216f3cc28d82cf7071981
> ```

Awesome! It works.

**Comment by dcroote** [comment]
@jiguantong - this feature is now available in the recently released [v0.11.2](https://github.com/api3dao/airnode/releases/tag/v0.11.2).
